iib/web/api_v1.py

```diff
@@ -217,17 +241,38 @@ def get_related_bundles(request_id):
             ' related_bundles will be available once the request is complete.'
         )
 
+    # If S3 bucket is configured, fetch the related buyndles file from the S3 bucket.
+    # Esle, check if related bundles are stored on the system itself and return them.
+    # Otherwise, raise an IIBError.
+    if s3_bucket_name:
+        related_bundles_file = get_file_from_s3_bucket(
+            f'related_bundles/{request_id}_related_bundles.json', s3_bucket_name
+        )
```

Comment:
Typos on `bundles` and `Else`

Another point is that it looks like you're repeating yourself for the keys `request_logs` and `related_bundles`.

The content inside this `if` looks similar to the other one above, just changing the `key_prefix` and whether is `text/plain` or `application/json`.

Maybe there's a more generic way to do that...

Reply:
I did think about this and was torn between the option of creating a helper function or not and decided against it. But now that you have mentioned it, I'll go ahead and do it ;)